src/bsp/platform/iotc_bsp_io_net_test.cc

```diff
@@ -0,0 +1,138 @@
+/* Copyright 2018-2019 Google LLC
+ *
+ * This is part of the Google Cloud IoT Device SDK for Embedded C,
+ * it is licensed under the BSD 3-Clause license; you may not use this file
+ * except in compliance with the License.
+ *
+ * You may obtain a copy of the License at:
+ *  https://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "iotc_bsp_io_net.h"
+
+#include <memory>
+#include <stdio.h>
+
+#include "gtest.h"
+#include "iotc_test_echoserver.h"
+
+namespace iotctest {
+namespace {
+
+typedef struct NetworkType_s {
+  iotc_bsp_socket_type_t socket_type;
+  iotc_bsp_protocol_type_t protocol_type;
+  std::string listening_addr;
+
+  NetworkType_s(iotc_bsp_socket_type_t socket_type_,
```

Comment:
Do not use `_`-suffixed parameter names. Just remove all suffixes and all should look fine.